rules/macos/credential_access_high_volume_of_pbpaste.toml

```diff
@@ -0,0 +1,120 @@
+   - **Why:** Understanding what is triggering `pbpaste` can help determine if this activity is legitimate or part of a malicious attempt to gather sensitive information.
+   - $investigate_1
+   - $investigate_2
+
+3. **Review Clipboard Contents:**
+   - **What to check:** If possible, capture and review the clipboard contents during `pbpaste` executions to identify if sensitive data, such as user credentials, is being targeted.
+   - **Why:** Attackers may use `pbpaste` to harvest valuable information from the clipboard. Identifying the type of data being collected can indicate the severity of the threat.
+
+4. **Check for Data Exfiltration:**
+   - **What to check:** Investigate any output files or network activity associated with `pbpaste` usage. Look for signs that the collected data is being saved to a file, transmitted over the network, or sent to an external location.
+   - **Why:** If data is being stored or transmitted, it may be part of an exfiltration attempt. Identifying this can help prevent sensitive information from being leaked.
+
+5. **Correlate with User Activity:**
+   - **What to check:** Compare the `pbpaste` activity with the user’s normal behavior and system usage patterns.
+   - **Why:** If the `pbpaste` activity occurs during times when the user is not active, or if the user denies initiating such tasks, it could indicate unauthorized access or a compromised account.
+
+By thoroughly investigating these aspects of `pbpaste` activity, you can determine whether this is part of a legitimate process or a potential security threat that needs to be addressed.
+"""
+risk_score = 47
+rule_id = "e29599ee-d6ad-46a9-9c6a-dc39f361890d"
+setup = """## Setup
+
+This rule requires data coming in from Jamf Protect.
+
+### Jamf Protect Integration Setup
+Jamf Protect is integrated into the Elastic Agent using Fleet. Upon configuration, the integration allows the Elastic Agent to monitor events incoming events and send data to the Elastic.
+
+#### Prerequisite Requirements:
+- Fleet is required for Jamf Protect.
+- To configure Fleet Server refer to the [documentation](https://www.elastic.co/guide/en/fleet/current/fleet-server.html).
+
+#### The following steps should be executed in order to add the Jamf Protect integration:
+- Go to the Kibana home page and click "Add integrations".
+- In the query bar, search for "Jamf Protect" and select the integration to see more details about it.
+- Click "Add Jamf Protect".
+- Configure the integration name.
+- Click "Save and Continue".
+"""
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: macOS",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+    "Data Source: Jamf Protect",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by host.hostname, host.id with maxspan=1m
```

Comment:
Should maxspan be reduced to maybe 5s or similar? 1m seems like a large time window that could be accomplished by an end user?

Reply:
It's not a end-user scenario we're trying to capture here but a pbpaste loop that runs like once 10s to capture input and write it out to a file.

Reply:
I think the question was if this is FP prone, as if the user could run the command in this 1m time window

Reply:
ah right - most typical end-users don't interact with `pbpaste` at all. it's either scripts invoking `pbpaste` or `pbcopy` or in some cases software invoking it, but i've not seen cases a end-user that triggers this with that threshold set in 1 minute.